runtime/src/accounts.rs

```diff
@@ -623,18 +678,17 @@ impl Accounts {
             {
                 if message.is_debitable(i) {
                     accounts.push((key, account));
-                }
-                if *credit > 0 {
-                    // Increment credit-only account balance Atomic
+                } else {
                     self.credit_only_account_locks
-                        .read()
+                        .write()
```

Comment:
why did this require a write lock when it didn't before?

Reply:
@rob-solana  before this, we only had to do `fetch_add` which doesnt require mutability. But since, now we are setting bool var, we need to have mutability, hence write lock

Reply:
how about an AtomicBool?

Reply:
hmm. Adding atomic bool, will enable us to use read lock only. But lock contention still occurs at that level. 

But, I see your point.